include/fastdds/dds/publisher/DataWriter.hpp

```diff
@@ -128,19 +128,19 @@ class DataWriter : public DomainEntity
      * Write data to the topic.
      *
      * @param data Pointer to the data
-     * @return True if correct, false otherwise
+     * @return RETCODE_OK if the data is correctly sent and RETCODE_ERROR otherwise.
```

Comment:
Since you changed also WriterImpl, now the method write() can return the specific error and not only RETCODE_ERROR
```suggestion
     * @return RETCODE_OK if the data is correctly sent or a ReturnCode related to the specific error otherwise.
```